src/main/kotlin/com/cognifide/gradle/aem/environment/io/Listener.kt

```diff
@@ -0,0 +1,31 @@
+package com.cognifide.gradle.aem.environment.io
+
+import java.io.File
+import org.apache.commons.io.monitor.FileAlterationListener
+import org.apache.commons.io.monitor.FileAlterationObserver
+
+class Listener(private val notify: (String) -> Unit) : FileAlterationListener {
+    override fun onDirectoryDelete(directory: File?) {
+        notify("${directory?.name} deleted")
```

Comment:
event should not be fired with null; IntelliJ automatically generates for you types with `?` because it does not if there will be null possible coming from 3rd party library. if we are sure about that null-safety, you could definitelly remove these `?`; I am quite sure that you could do this

btw, what is more `notify(" created")` will be not interesting if null magically be passed here :D you should avoid such notification :D